File: GFG/BST/leftceil.py
import ds

def lceil(arr):
    bt=ds.BST()
    for i in arr:
        print(bt.ceil(i).key,end=" ") if(bt.ceil(i)!=None) else print(-1,end=" ")
        bt.insert(i)
        
arr=[2,8,30,15,25,12]
lceil(arr)
    

File: GFG/BST/leftceil.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

void lceil(vector<int> arr){
    BST<int> bt;
    
    for(auto j:arr){
        (bt.ceil(j)!=NULL)?cout<<bt.ceil(j)->key<<" " : cout<<-1<<" ";
        bt.insert(j);
    }
}
int main(){
    vector<int> arr{2,8,30,15,25,12};
    lceil(arr);
    return 0;
}